Adding clarification about the agent being started via ssh. (#5104)
jenkins-infra/jenkins.io: content/doc/book/using/using-agents.adoc

=== Generating an SSH key pair

The SSH key pair generation can be done on any computer, whatever its OS is (Linux, Windows, MacOs, ...). On Windows, you could use https://cygwin.com/[Cygwin] for example. You could do it on your Jenkins controller machine, but also on the host that will become an agent, on your laptop, or any other machine, as long as you have access to the `ssh-keygen` command.
Note that you will have to be able to copy the key value to your controller and agent afterwards, so you check that you can copy a file content into the clipboard beforehand.

1. In a terminal window run the command: `ssh-keygen -f ~/.ssh/jenkins_agent_key`

> 2. Provide a passphrase to use with the key (it can be empty, even if that's not such a good idea)

Reviewer: agree please don't add 'even if that's not such a good idea)'

> 2. Provide a passphrase to use with the key (it can be empty, even if that's not such a good idea)

Author: Got it, thanks!

=== Generating an SSH key pair

To generate the SSH key pair, you have to execute a command line tool named `ssh-keygen` on a machine you have access to. It could be:

 * the machine on which your Jenkins controller runs
 * the host (if using containers)
 * a machine on which you have an agent running
 * or even your developer machine

[NOTE]
====
The SSH key pair generation can be done on any operating system:

> * On Windows, you can use any OpenSSH installation such as https://docs.microsoft.com/en-us/windows-server/administration/openssh/openssh_install_firstuse[Windows OpenSSH] or https://cygwin.com/[Cygwin]

Reviewer 2: ```suggestion
* On Windows, you can use any OpenSSH installation such as https://docs.microsoft.com/en-us/windows-server/administration/openssh/openssh_install_firstuse[Windows OpenSSH], the `ssh-keygen` that is included with https://gitforwindows.org/[git for Windows], or https://cygwin.com/[Cygwin]
```

>  * On macOS or Unix/Linux, you can use any OpenSSH installation packaged with your system as well

Reviewer 2: ```suggestion
 * On Unix (Linux, macOS, BSD, etc.) you can use any OpenSSH installation packaged with your system as well
```

[NOTE]
====
* Remember to replace the tag [your-public-key] for your own SSH *public* key.

> * Your public key value in this example could be found by issuing : `cat ~/.ssh/jenkins_agent_key.pub` on the machine your created it. Do not add the square brackets `[]` around the key value

Author: The square brackets come from the command line [example](https://deploy-preview-5104--jenkins-io-site-pr.netlify.app/doc/book/using/using-agents/#creating-your-docker-agent):
` docker run -d --rm --name=agent1 -p 22:22 \
-e "JENKINS_AGENT_SSH_PUBKEY=[your-public-key]" \
jenkins/ssh-agent:alpine` 

It's a no-brainer, no one should ever have the idea to add brackets surrounding the key, except... I did it when my first try with my public key did not work (spoiler alert, it didn't work either with the brackets). :-D

> * Your public key value in this example could be found by issuing : `cat ~/.ssh/jenkins_agent_key.pub` on the machine your created it. Do not add the square brackets `[]` around the key value

Reviewer 3: Make sense, thanks for the explanation!

+
image:node/node-4.png[Agent successfully connected,650]

> If your Jenkins controller does not manage to start the agent via ssh, please check the port you <<ssh-anchor,configured>> on your agent. Copy it, and then click on the `Advanced...` button. You will then be able to paste the port number into the `Port` textfield.

Reviewer 2: Site prefers sentence per line format for easier review of changes.
```suggestion
If your Jenkins controller does not start the agent via ssh, please check the port you <<ssh-anchor,configured>> on your agent.
Copy it, and then click on the `Advanced...` button.
You will then be able to paste the port number into the `Port` textfield.
```
